### ISSUE TITLE
java.lang.IllegalStateException: trailers not available

### ISSUE THREAD

**Issue by ln-12** [connectrpc/connect-kotlin#259]
Hey, I am using the Connect library in my app which works pretty good so far. We have to stick to v0.5.0 for now as some other dependency need protobuf 3.25 which makes it incompatible with 0.6.0 for now.

However, I get some errors from users via Sentry like this:

```
java.lang.IllegalStateException: trailers not available
    at okhttp3.Response.trailers(Response.kt:180)
    at com.connectrpc.okhttp.ConnectOkHttpClient$unary$1.onResponse(ConnectOkHttpClient.kt:160)
    at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
    at java.lang.Thread.run(Thread.java:1012)
```

This seems to be the case whenever the network connection is lost / flaky.

I am executing the call to the network function inside a normal try catch block. The stacktrace also suggests that I cannot wrap it in a way that catches the crash as it directly originates from the OkHttp threadpool rather than my code.

Can you have a look at it please? And maybe you have an idea for a quick workaround as this occurs quite regularly.

**Comment by ln-12** [comment]
After looking at the current implementation and copying it to my project it seems like this is fixed in https://github.com/connectrpc/connect-kotlin/commit/16a8b1dc8ba18a2d1fcafa42e87ffe2c6a0bee7f already.

**Comment by pkwarren** [comment]
Hi @ln-12 - we've published [v0.5.1](https://github.com/connectrpc/connect-kotlin/releases/tag/v0.5.1) (for protobuf-java 3.x) and [v0.6.1](https://github.com/connectrpc/connect-kotlin/releases/tag/v0.6.1) (for protobuf-java 4.x) with this fix.

**Comment by ln-12** [comment]
Awesome! Thank you very much 👍
